networkx/algorithms/centrality/group.py

```diff
@@ -515,7 +513,7 @@ def _heuristic(k, root, DF_tree, D, nodes, greedy):
             zip(np.diag(DF_tree.nodes[node_p]["betweenness"]), nodes), reverse=True
         )
     ]
-    [CL.remove(m) for m in CL if m in DF_tree.nodes[node_p]["GM"]]
+    _ = [CL.remove(m) for m in CL if m in DF_tree.nodes[node_p]["GM"]]
```

Comment:
This change (I find) makes it even harder to read than the poor readable version we had before.
Can we just use:
```suggestion
    added_nodes = DF_tree.nodes[node_p]["GM"]
    for m in CL:
        if m in added_nodes:
            CL.remove(m)  
```

This big loop could be sped up by preassigning `DF_tree.nodes[node_p]` to a variable and maybe `DF_tree.nodes[node_p]["GM"]` and `root_node["sigma"]`.  Maybe that's out of scope for this PR, but we are looking at this code, so if not here, we should make an issue to update this code.